pkg/cli/zip.go

```diff
@@ -579,7 +580,12 @@ func runDebugZip(cmd *cobra.Command, args []string) (retErr error) {
 			} else {
 				fmt.Printf("%d found\n", len(profiles.Files))
 				for _, file := range profiles.Files {
-					name := prefix + "/heapprof/" + file.Name + ".pprof"
+					name := prefix + "/heapprof/" + file.Name
```

Comment:
It seems weird that we're massaging the file endings here. Wouldn't we rather write these files using the correct endings (in which case nothing to do here) instead?
If it's too late in the cycle for that and this is a band-aid, mention that in the PR. (I think it's not too late in the cycle, and given that we did change the names in this cycle we should be free to change them again last minute).

Reply:
Good points.

Generally we want file extensions otherwise it's a chore for the user to understand what to do.
I agree this is hacky and this should be best done server-side upfront when the file is created.
I changed the PR to do this. PTAL.

Even after we fix this at the source (i.e. inside the server) we still need this hack in `zip.go` for backward-compatibility with previous servers. Added a comment to explain.